src/DataLib/fendl32B_retrofit/reaction_data.py

```diff
@@ -134,5 +137,76 @@ def load_mt_table(csv_path):
         csv_reader = csv.DictReader(f)
         for row in csv_reader:
             mt_dict[row['MT']] = {'Reaction' : row['Reaction']}
+    if mt_dict:
+        return mt_dict
+    else:
+        raise Exception('CSV file is empty or missing.')
+
+def check_for_isomer(emitted_particle_string):
+    """
+    Check the isomeric status of a neutron-activated nucleus.
+        By the formatting conventions of ENDF reaction types,
+        if the string of a reaction product ends with a digit,
+        that signifies the excitation state of the nucleus, so 
+        this function looks for and stores these values.
+
+    Arguments:
+        emitted_particle_string (str): Particle product(s) of the neutron
+            activation.
+    
+    Returns:
+        isomeric_value (int): Nuclear excitation level of the activated
+            nucleus. For a nucleus in the ground state, isomeric_state = 0.
+    """
+
+    last_digits_str = ''
+    for char in reversed(emitted_particle_string):
+        if char.isdigit():
+            last_digits_str = char + last_digits_str
+        else:
+            break
+    isomeric_value = int(last_digits_str) if last_digits_str else 0
 
+    return isomeric_value
+
+def process_mt_data(mt_dict):
+    """
+    Read in the dictionary containing the data from mt_table.csv that is
+        imported using load_mt_data() and process the reaction data to 
+        tally the emitted particles by their respective types and calculate
+        the change in KZA associated with each reaction. Write out these
+        reaction-specific values to new keys in the dictionary.
+    
+    Arguments:
+        mt_dict (dict): Dictionary formatted data structure for mt_table.csv,
+            with the general format: {'MT' : {'Reaction' : (z , emission)}}.
+    
+    Returns:
+        mt_dict (dict): Processed dictionary containing the original data from
+            the dictionary formatted mt_table.csv, with additional information
+            on changes in KZA and emitted particles. The returned format is:
+            {'MT' : 
+                    {'Reaction'         :               (z, emission)},
+                    {'delKZA'           :       integer change in KZA},
+                    {'Emitted Particles : string of emitted particles}
+            }
+    """
+
+    for MT, data in mt_dict.items():
+        emitted_particles = data['Reaction'].split(',')[1][:-1]
+        emission_dict = emission_breakdown(emitted_particles)
+        change_NP = list(nucleon_changes(emission_dict))
```

Comment:
I don't think you need to save this as a list, with the above change, it's either a numpy array or `None`
```suggestion
        change_NP = nucleon_changes(emission_dict)
```